api/app/src/command/medical/document/document-query.ts

```diff
@@ -138,6 +155,38 @@ export const updateConversionProgress = async ({
     };
     await PatientModel.update(updatedPatient, { where: patientFilter, transaction });
 
+    // START TODO 785 remove this once we're confident with the flow
+    const maxUpdateRetry = 3;
+    let curAttempt = 1;
+    while (curAttempt++ < maxUpdateRetry) {
+      const patientPost = await getPatientOrFail({
+        id: patient.id,
+        cxId: patient.cxId,
+        transaction,
+      });
+      log(
+        `[txn attempt ${curAttempt}] Status post-update: ${JSON.stringify(
+          patientPost.data.documentQueryProgress
+        )}`
+      );
+      if (
+        !isDocumentQueryProgressEqual(documentQueryProgress, patientPost.data.documentQueryProgress)
+      ) {
+        log(`[txn attempt ${curAttempt}] Status post-update not expected... trying again`);
+        await PatientModel.update(updatedPatient, { where: patientFilter, transaction });
+      } else {
+        log(`[txn attempt ${curAttempt}] Status post-update is as expected!`);
+        break;
+      }
+    }
+    const patientPost = await getPatientOrFail({
+      id: patient.id,
+      cxId: patient.cxId,
+      transaction,
+    });
```

Comment:
We could remove this and add the log right below to the successful loop exit - or expose `patientPost` to outside of the loop

Reply:
I guess we don't need this here anyways, since the intention was to capture the status after the last update attempt - however the code in `/doc-conversion-status` will take care of that.

Will remove this.